<Type Name="ThreadInterruptedException" FullName="System.Threading.ThreadInterruptedException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="362c5c18ecdef46e8c16efe60b595b9ca075f963" /><Meta Name="ms.sourcegitcommit" Value="b6b9d5f3f9c7eb43f14a8a1a68d044603cb8ed73" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ja-JP" /><Meta Name="ms.lasthandoff" Value="05/18/2020" /><Meta Name="ms.locfileid" Value="83480872" /></Metadata><TypeSignature Language="C#" Value="public class ThreadInterruptedException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ThreadInterruptedException extends System.SystemException" FrameworkAlternate="net-5.0;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netcore-3.1;netstandard-2.0;netstandard-2.1" />
  <TypeSignature Language="DocId" Value="T:System.Threading.ThreadInterruptedException" />
  <TypeSignature Language="VB.NET" Value="Public Class ThreadInterruptedException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ThreadInterruptedException : SystemException" />
  <TypeSignature Language="F#" Value="type ThreadInterruptedException = class&#xA;    inherit SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit ThreadInterruptedException extends System.SystemException" FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Threading.Thread</AssemblyName>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
    <AssemblyVersion>4.1.2.0</AssemblyVersion>
    <AssemblyVersion>5.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <TypeForwardingChain>
    <TypeForwarding From="netstandard" FromVersion="2.1.0.0" To="System.Threading.Thread" ToVersion="5.0.0.0" FrameworkAlternate="net-5.0" />
  </TypeForwardingChain>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="1818d-101"><see cref="T:System.Threading.Thread" /> が待機状態のときに中断されるとスローされる例外。</span><span class="sxs-lookup"><span data-stu-id="1818d-101">The exception that is thrown when a <see cref="T:System.Threading.Thread" /> is interrupted while it is in a waiting state.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1818d-102">スレッドが作成された後は、破棄されるまで、1つまたは複数の <xref:System.Threading.ThreadState> 状態になります。</span><span class="sxs-lookup"><span data-stu-id="1818d-102">After a thread is created, it is in one or more <xref:System.Threading.ThreadState> states until it is destroyed.</span></span> <span data-ttu-id="1818d-103">スレッドが <xref:System.Threading.ThreadState.WaitSleepJoin> 状態にあるときに <xref:System.Threading.Thread.Interrupt%2A> を呼び出すと、ターゲットスレッドで <xref:System.Threading.ThreadInterruptedException> がスローされます。</span><span class="sxs-lookup"><span data-stu-id="1818d-103">Calling <xref:System.Threading.Thread.Interrupt%2A> when a thread is in the <xref:System.Threading.ThreadState.WaitSleepJoin> state will cause a <xref:System.Threading.ThreadInterruptedException> to be thrown in the target thread.</span></span> <span data-ttu-id="1818d-104">スレッドが <xref:System.Threading.ThreadState.WaitSleepJoin> 状態でない場合、スレッドがその状態になるまで例外はスローされません。</span><span class="sxs-lookup"><span data-stu-id="1818d-104">If the thread is not in the <xref:System.Threading.ThreadState.WaitSleepJoin> state, the exception is not thrown until the thread enters that state.</span></span> <span data-ttu-id="1818d-105">スレッドがまったくブロックされない場合は、中断されることなく完了することがあります。</span><span class="sxs-lookup"><span data-stu-id="1818d-105">If the thread never blocks, it could complete without ever being interrupted.</span></span>  
  
 <span data-ttu-id="1818d-106"><xref:System.Threading.ThreadInterruptedException> では、値0x80131519 の HRESULT COR_E_THREADINTERRUPTED を使用します。</span><span class="sxs-lookup"><span data-stu-id="1818d-106"><xref:System.Threading.ThreadInterruptedException> uses the HRESULT COR_E_THREADINTERRUPTED, which has the value 0x80131519.</span></span>  
  
 <span data-ttu-id="1818d-107"><xref:System.Threading.ThreadInterruptedException>インスタンスの初期プロパティ値の一覧については、<xref:System.Threading.ThreadInterruptedException.%23ctor%2A>コンストラクターを参照してください。</span><span class="sxs-lookup"><span data-stu-id="1818d-107">For a list of initial property values for an instance of <xref:System.Threading.ThreadInterruptedException>, see the <xref:System.Threading.ThreadInterruptedException.%23ctor%2A> constructors.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="1818d-108">次のコード例は、実行中のスレッドが中断され、その後ブロックされた場合の動作を示しています。</span><span class="sxs-lookup"><span data-stu-id="1818d-108">The following code example shows the behavior of a running thread when it is interrupted and subsequently gets blocked.</span></span>  
  
 [!code-cpp[System.Threading.Thread.Interrupt#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Threading.Thread.Interrupt/CPP/source.cpp#1)]
 [!code-csharp[System.Threading.Thread.Interrupt#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Threading.Thread.Interrupt/CS/source.cs#1)]
 [!code-vb[System.Threading.Thread.Interrupt#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Threading.Thread.Interrupt/VB/source.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Threading.Thread" />
    <altmember cref="T:System.Threading.ThreadState" />
    <related type="Article" href="~/docs/standard/threading/pausing-and-resuming-threads.md"><span data-ttu-id="1818d-109">スレッドの一時中断と再開</span><span class="sxs-lookup"><span data-stu-id="1818d-109">Pausing and Resuming Threads</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="1818d-110"><see cref="T:System.Threading.ThreadInterruptedException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="1818d-110">Initializes a new instance of the <see cref="T:System.Threading.ThreadInterruptedException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ThreadInterruptedException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.ThreadInterruptedException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ThreadInterruptedException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
        <AssemblyVersion>5.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="1818d-111"><see cref="T:System.Threading.ThreadInterruptedException" /> クラスの新しいインスタンスを、既定のプロパティを使用して初期化します。</span><span class="sxs-lookup"><span data-stu-id="1818d-111">Initializes a new instance of the <see cref="T:System.Threading.ThreadInterruptedException" /> class with default properties.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1818d-112">次の表に、<xref:System.Threading.ThreadInterruptedException> のインスタンスに対するプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="1818d-112">The following table shows the initial property values for an instance of <xref:System.Threading.ThreadInterruptedException>.</span></span>  
  
|<span data-ttu-id="1818d-113">プロパティ</span><span class="sxs-lookup"><span data-stu-id="1818d-113">Property</span></span>|<span data-ttu-id="1818d-114">値</span><span class="sxs-lookup"><span data-stu-id="1818d-114">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1818d-115">[https://login.microsoftonline.com/consumers/](`null`)</span><span class="sxs-lookup"><span data-stu-id="1818d-115">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1818d-116">ローカライズされたエラー メッセージの文字列です。</span><span class="sxs-lookup"><span data-stu-id="1818d-116">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/threading/pausing-and-resuming-threads.md"><span data-ttu-id="1818d-117">スレッドの一時中断および中断</span><span class="sxs-lookup"><span data-stu-id="1818d-117">Pausing and interrupting threads</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ThreadInterruptedException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.ThreadInterruptedException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ThreadInterruptedException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Threading.ThreadInterruptedException : string -&gt; System.Threading.ThreadInterruptedException" Usage="new System.Threading.ThreadInterruptedException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
        <AssemblyVersion>5.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="1818d-118">例外の原因を示すエラー メッセージです。</span><span class="sxs-lookup"><span data-stu-id="1818d-118">The error message that explains the reason for the exception.</span></span></param>
        <summary><span data-ttu-id="1818d-119">指定したエラー メッセージを使用して、<see cref="T:System.Threading.ThreadInterruptedException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="1818d-119">Initializes a new instance of the <see cref="T:System.Threading.ThreadInterruptedException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1818d-120">次の表に、<xref:System.Threading.ThreadInterruptedException> のインスタンスに対するプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="1818d-120">The following table shows the initial property values for an instance of <xref:System.Threading.ThreadInterruptedException>.</span></span>  
  
|<span data-ttu-id="1818d-121">プロパティ</span><span class="sxs-lookup"><span data-stu-id="1818d-121">Property</span></span>|<span data-ttu-id="1818d-122">値</span><span class="sxs-lookup"><span data-stu-id="1818d-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1818d-123">[https://login.microsoftonline.com/consumers/](`null`)</span><span class="sxs-lookup"><span data-stu-id="1818d-123">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1818d-124">エラー メッセージ文字列です。</span><span class="sxs-lookup"><span data-stu-id="1818d-124">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/threading/pausing-and-resuming-threads.md"><span data-ttu-id="1818d-125">スレッドの一時中断と再開</span><span class="sxs-lookup"><span data-stu-id="1818d-125">Pausing and Resuming Threads</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected ThreadInterruptedException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; ThreadInterruptedException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Threading.ThreadInterruptedException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Threading.ThreadInterruptedException" Usage="new System.Threading.ThreadInterruptedException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
        <AssemblyVersion>5.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="1818d-126">スローされた例外に関する、シリアル化されたオブジェクト データを保持する <see cref="T:System.Runtime.Serialization.SerializationInfo" /> です。</span><span class="sxs-lookup"><span data-stu-id="1818d-126">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</span></span></param>
        <param name="context"><span data-ttu-id="1818d-127">転送元または転送先についてのコンテキスト情報を含む <see cref="T:System.Runtime.Serialization.StreamingContext" /> です。</span><span class="sxs-lookup"><span data-stu-id="1818d-127">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="1818d-128">シリアル化されたデータを使用して、<see cref="T:System.Threading.ThreadInterruptedException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="1818d-128">Initializes a new instance of the <see cref="T:System.Threading.ThreadInterruptedException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1818d-129">このコンストラクターは、シリアル化解除の際に呼び出されて、ストリームで送信された例外オブジェクトを再構成します。</span><span class="sxs-lookup"><span data-stu-id="1818d-129">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="1818d-130">詳細については、次を参照してください。 [XML および SOAP シリアル化](~/docs/standard/serialization/xml-and-soap-serialization.md)します。</span><span class="sxs-lookup"><span data-stu-id="1818d-130">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/threading/pausing-and-resuming-threads.md"><span data-ttu-id="1818d-131">スレッドの一時中断と再開</span><span class="sxs-lookup"><span data-stu-id="1818d-131">Pausing and Resuming Threads</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ThreadInterruptedException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ThreadInterruptedException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Threading.ThreadInterruptedException : string * Exception -&gt; System.Threading.ThreadInterruptedException" Usage="new System.Threading.ThreadInterruptedException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.1.2.0</AssemblyVersion>
        <AssemblyVersion>5.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="1818d-132">例外の原因を示すエラー メッセージです。</span><span class="sxs-lookup"><span data-stu-id="1818d-132">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="1818d-133">現在の例外の原因になった例外です。</span><span class="sxs-lookup"><span data-stu-id="1818d-133">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="1818d-134"><paramref name="innerException" /> パラメーターが <see langword="null" /> でない場合は、内部例外を処理する <see langword="catch" /> ブロックで現在の例外が発生します。</span><span class="sxs-lookup"><span data-stu-id="1818d-134">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="1818d-135">指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を使用して、<see cref="T:System.Threading.ThreadInterruptedException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="1818d-135">Initializes a new instance of the <see cref="T:System.Threading.ThreadInterruptedException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1818d-136">前回の例外の直接的な結果としてスローされる例外は、<xref:System.Exception.InnerException%2A> プロパティに前回の例外への参照を含める必要があります。</span><span class="sxs-lookup"><span data-stu-id="1818d-136">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="1818d-137"><xref:System.Exception.InnerException%2A> プロパティは、コンストラクターに渡されるのと同じ値を返します。または、<xref:System.Exception.InnerException%2A> プロパティがコンストラクターに内部例外値を提供しない場合には `null` を返します。</span><span class="sxs-lookup"><span data-stu-id="1818d-137">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="1818d-138">次の表に、<xref:System.Threading.ThreadInterruptedException> のインスタンスに対するプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="1818d-138">The following table shows the initial property values for an instance of <xref:System.Threading.ThreadInterruptedException>.</span></span>  
  
|<span data-ttu-id="1818d-139">プロパティ</span><span class="sxs-lookup"><span data-stu-id="1818d-139">Property</span></span>|<span data-ttu-id="1818d-140">値</span><span class="sxs-lookup"><span data-stu-id="1818d-140">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="1818d-141">内部例外の参照です。</span><span class="sxs-lookup"><span data-stu-id="1818d-141">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="1818d-142">エラー メッセージ文字列です。</span><span class="sxs-lookup"><span data-stu-id="1818d-142">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/threading/pausing-and-resuming-threads.md"><span data-ttu-id="1818d-143">スレッドの一時中断と再開</span><span class="sxs-lookup"><span data-stu-id="1818d-143">Pausing and Resuming Threads</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>
